modules/representation_management/app/models/representation_management/form_2122_base.rb

```diff
@@ -0,0 +1,82 @@
+      veteran_social_security_number
+      veteran_file_number
+      veteran_address_line1
+      veteran_address_line2
+      veteran_city
+      veteran_country
+      veteran_state_code
+      veteran_zip_code
+      veteran_zip_code_suffix
+      veteran_area_code
+      veteran_phone_number
+      veteran_email
+      veteran_service_number
+      veteran_insurance_numbers
+    ]
+
+    claimant_attrs = %i[
+      claimant_first_name
+      claimant_middle_initial
+      claimant_last_name
+      claimant_address_line1
+      claimant_address_line2
+      claimant_city
+      claimant_country
+      claimant_state_code
+      claimant_zip_code
+      claimant_zip_code_suffix
+      claimant_area_code
+      claimant_phone_number
+      claimant_email
+      claimant_relationship
+    ]
+
+    consent_attrs = %i[
+      record_consent
+      consent_address_change
+      consent_limits
+    ]
+
+    validates :veteran_first_name, presence: true, length: { maximum: 12 }
+    validates :veteran_middle_initial, length: { maximum: 1 }
+    validates :veteran_last_name, presence: true, length: { maximum: 18 }
+    validates :veteran_social_security_number, presence: true, lenghth: { is: 9 }, numericality: { only_integer: true }
+    validates :veteran_file_number, presence: true, length: { is: 9 }, numericality: { only_integer: true }
+    validates :veteran_address_line1, presence: true, length: { maximum: 30 }
+    validates :veteran_address_line2, length: { maximum: 5 }
+    validates :veteran_city, presence: true, length: { maximum: 18 }
+    validates :veteran_country, presence: true, length: { is: 2 }
+    validates :veteran_state_code, presence: true, length: { is: 2 }
+    validates :veteran_zip_code, presence: true, length: { is: 5 }, numericality: { only_integer: true }
+    validates :veteran_zip_code_suffix, length: { is: 4 }, numericality: { only_integer: true }
+    validates :veteran_area_code, length: { is: 3 }, numericality: { only_integer: true }
+    validates :veteran_phone_number, length: { is: 7 }, numericality: { only_integer: true }
+    validates :veteran_service_number, length: { is: 9 }, numericality: { only_integer: true }
+    validates :veteran_insurance_numbers
+
+    with_options if: claimant_first_name.present? do
+      validates :claimant_first_name, presence: true, length: { maximum: 12 }
+      validates :claimant_middle_initial, length: { maximum: 1 }
+      validates :claimant_last_name, presence: true, length: { maximum: 18 }
+      validates :claimant_address_line1, presence: true, length: { maximum: 30 }
+      validates :claimant_address_line2, length: { maximum: 5 }
+      validates :claimant_city, presence: true, length: { maximum: 18 }
+      validates :claimant_country, presence: true, length: { is: 2 }
+      validates :claimant_state_code, presence: true, length: { is: 2 }
+      validates :claimant_zip_code, presence: true, length: { is: 5 }, numericality: { only_integer: true }
```

Comment:
I think this works, but it looks a bit off since zip codes are usually strings (they can start with 0). It might be clearer to do it this way:

`validates :claimant_zip_code, presence: true, format: { with: /\A\d{5}\z/, message: "must be 5 digits" }`

This way, we’re treating ZIP codes as strings and making sure they’re exactly 5 digits, which feels more natural.

Same for the other zip and phone validations.

Reply:
This is a great point.  The validations are updated to use this method.